content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java

```diff
@@ -723,10 +723,12 @@ public News scheduleNews(News news, Identity currentIdentity, String newsObjectT
       updateNewsArticle(news, currentIdentity, NewsUtils.NewsUpdateType.SCHEDULE.name().toLowerCase());
     }
     if (news != null) {
-      if (news.isPublished()) {
-        publishNews(news, currentIdentity.getUserId());
-      } else {
-        unpublishNews(news.getId(), currentIdentity.getUserId());
+      if (NewsUtils.canPublishNews(space.getId(), currentIdentity)) {
+        if (news.isPublished()) {
+          publishNews(news, currentIdentity.getUserId());
+        } else {
+          unpublishNews(news.getId(), currentIdentity.getUserId());
+        }
```

Comment:
why this is changed ?

Reply:
to handle unauthorized exception when the user is a space redactor and not a space publisher !